src/QirRuntime/lib/QIR/intrinsicsMath.cpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <cmath>
+#include "quantum__qis.hpp"
+
+extern "C"
+{
+
+// Implementations:
+bool quantum__qis__isnan__body(double d)
+{
+    return std::isnan(d);           // https://en.cppreference.com/w/cpp/numeric/math/isnan
+}
+
+double quantum__qis__infinity__body()
+{
+    return INFINITY;                // https://en.cppreference.com/w/c/numeric/math/INFINITY
+}
+
+bool quantum__qis__isinf__body(double d)
+{
+    return std::isinf(d);           // https://en.cppreference.com/w/cpp/numeric/math/isinf
+}
+
+double quantum__qis__arctan2__body(double y, double x)
+{
+    return std::atan2(y, x);        // https://en.cppreference.com/w/cpp/numeric/math/atan2
```

Comment:
As per your note below, does this introduce a discrepancy with existing semantics for Q#'s `ArcTan2` function?

Reply:
See [this my reply](https://github.com/microsoft/qsharp-runtime/pull/500#discussion_r572467750).